### ISSUE TITLE
I can't do search and replace for multiple lines of code

### ISSUE THREAD

**Issue by me-suzy** [stefankueng/grepWin#460]
**I try to FIND and REPLACE this:**

```
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-5LXZSBG');</script>
<!-- End Google Tag Manager -->
```

**with this:**

```
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-5QVRTGE6B7"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-5QVRTGE6B7');
</script>
<!-- End Google Tag Manager -->
```

This is a simple find and replace. Doesn't work at all. I also try with regex, I try all combinations. Doesn't work

![image](https://github.com/stefankueng/grepWin/assets/2770489/45eaee96-b04d-4419-b02e-65630600546c)

**or this:**

![image](https://github.com/stefankueng/grepWin/assets/2770489/3a5f4708-2abc-4aef-a6fe-7c0e7927550e)

**Comment by lifenjoiner** [comment]
Will be fixed after I finish the code refactoring at hand. There are conflicts.
Thanks for your report!

**Comment by me-suzy** [comment]
this problem was from the beginning. Please write me back after you repair the code. Please test the code with other variables. Because I may have longer text, with any kinds of simbols, regex, etc. Regular TEXT SEARCH mus FIND and REPLACE all the lines exactly as they are

**Comment by lifenjoiner** [comment]
try [bf47a2a](https://github.com/stefankueng/grepWin/commit/bf47a2aa81cc471db12f477fc76d4ee3a27dbd2a)

**Comment by me-suzy** [comment]
how to use that code?? I am frontend use. I must use directly the software, not the code...

I am using this version, portable:

  [grepWin-2.0.15_portable.zip](https://github.com/stefankueng/grepWin/releases/download/2.0.15/grepWin-2.0.15_portable.zip)

**Comment by stefankueng** [comment]
https://github.com/stefankueng/grepWin/actions/runs/8065122982
at the bottom, click the 'grepwin-64' link:
https://github.com/stefankueng/grepWin/actions/runs/8065122982/artifacts/1278716808

**Comment by me-suzy** [comment]
works, thanks. You can now update the newest verstion with this one.

![image](https://github.com/stefankueng/grepWin/assets/2770489/7749d8a8-d19b-4356-b894-eb5971381603)

**Comment by me-suzy** [comment]
also be aware of one thing. In this version, if I check the option **"Include Subfolders"**, than it is not possible to do FIND and REPLACE **in several folders.**

For example SEARCH in two folders, and I check the option "Include subfolders"

`e:\Carte\BB\17 - Site Leadership\Principal\ro\|e:\Carte\BB\17 - Site Leadership\Principal\en\`

It doesn't do anything.

I have to take a separate folder:

e:\Book\BB\17 - Site Leadership\Principal\ro\
e:\Book\BB\17 - Site Leadership\Principal\en\

If I don't check the option "Include subfolders", then it works to FIND and REPLACE in many folders

**Comment by lifenjoiner** [comment]
> several folders

fixed by [7562c1f](https://github.com/stefankueng/grepWin/commit/7562c1f00310b16e2b530f170f019125dad23ddf).

**Comment by me-suzy** [comment]
ok, these new updates, will be in the newest version?

**Comment by lifenjoiner** [comment]
There are other works in progress. There will be new version when it is stable enough. Stay steady.

**Comment by lifenjoiner** [comment]
There are other works in progress. There will be new version when it is stable enough. Stay steady.

**Comment by me-suzy** [comment]
please send me the .exe files, with the last fixed by [7562c1f](https://github.com/stefankueng/grepWin/commit/7562c1f00310b16e2b530f170f019125dad23ddf).

I want to test again, see if replacement in subfolders works
